cfpb/cfgov-refresh
# Add schema_json field to CFGOVPage for structured data

This field can be used to store structured data about a page, which is then output in the `<head>`, between `<script type="application/ld+json">` and `</script>` tags.

Such structured data can be used by search engines to provide "rich" 
results when these pages are returned by a search.

Learn more at: https://developers.google.com/search/docs/guides/intro-structured-data

Part of addressing https://JIRA/browse/INTAKE-4248


## Additions

- `schema_json` field added to `CFGOVPage` model with output in the `<head>` of a page

## Testing

1. Pull branch
1. Visit any page in the browser, view the source, and verify that you cannot find `<script type="application/ld+json">`
1. Open up the page in the Admin and go to its Configuration tab
1. Enter valid JSON into the Schema JSON field and publish the page
1. Reload the rendered page and look for the `<script type="application/ld+json">` in the source, with the JSON you entered

## Notes

- When the field is empty, the word `null` appears in it. This doesn't get output on the page, but it's still somewhat weird. Any advice would be welcome.
- Also hoping to get some help from the Wagtail Slack on retaining line breaks that are entered into the field.

## Todos

- [x] Add tests for new context variable

## Checklist
